static/usage/popover/presenting/controller/react.md

```diff
@@ -0,0 +1,24 @@
+```tsx
+import React, { useState } from 'react';
+import { IonButton, IonContent, useIonPopover } from '@ionic/react';
+
+const Popover = () => <IonContent className="ion-padding">Hello World!</IonContent>;
+
+function Example() {
+  const [present, dismiss] = useIonPopover(Popover, {
+    onDismiss: (data, role) => dismiss(data, role)
+  });
+  const [roleMsg, setRoleMsg] = useState('');
+
+  return (
+    <div>
+      <IonButton onClick={(e: any) => present({
+        event: e,
+        onDidDismiss: (e: any) => setRoleMsg(`Popover dismissed with role: ${e.detail.role}`)
```

Comment:
Can we add types here?

Reply:
Also this example does not compile on StackBlitz.

```
ERROR in src/main.tsx:8:17
TS7006: Parameter 'data' implicitly has an 'any' type.
     6 | function Example() {
     7 |   const [present, dismiss] = useIonPopover(Popover, {
  >  8 |     onDismiss: (data, role) => dismiss(data, role)
       |                 ^^^^
     9 |   });
    10 |   const [roleMsg, setRoleMsg] = useState('');
    11 |

ERROR in src/main.tsx:8:23
TS7006: Parameter 'role' implicitly has an 'any' type.
     6 | function Example() {
     7 |   const [present, dismiss] = useIonPopover(Popover, {
  >  8 |     onDismiss: (data, role) => dismiss(data, role)
       |                       ^^^^
     9 |   });
    10 |   const [roleMsg, setRoleMsg] = useState('');
    11 |

ERROR in src/main.tsx:8:40
TS2554: Expected 0 arguments, but got 2.
     6 | function Example() {
     7 |   const [present, dismiss] = useIonPopover(Popover, {
  >  8 |     onDismiss: (data, role) => dismiss(data, role)
       |                                        ^^^^^^^^^^
     9 |   });
    10 |   const [roleMsg, setRoleMsg] = useState('');
    11 |
```

Reply:
Do you know what the type for the `onClick` event should be? Looks like it's typically `React.MouseEvent`, but I'm unable to cast it to `Event` to pass it to `present`. I could do `event: e as any` I suppose, but that kind of defeats the purpose haha.

The error with `dismiss` getting 2 arguments is handled by the dev build in the PR description.

Reply:
You could get really specific and do `PointerEvent`. Otherwise `UIEvent` may work.

Reply:
`onClick` still complains, since it expects a `MouseEvent`. Worth noting that our current docs just use `any`: https://ionicframework.com/docs/api/popover#inline-popover-with-state

Reply:
`any` is fine then. There may be a type error in our wrappers somewhere.